tests/topology_test.go

```diff
@@ -837,3 +838,44 @@ func TestQueryMetadata(t *testing.T) {
 
 	RunTest(t, test)
 }
+
+//TestUserMetadata tests user metadata functionality
+func TestUserMetadata(t *testing.T) {
+	test := &Test{
+		setupCmds: []helper.Cmd{
+			{"ovs-vsctl add-br br-umd", true},
+		},
+
+		tearDownCmds: []helper.Cmd{
+			{"ovs-vsctl del-br br-umd", true},
+		},
+
+		check: func(c *TestContext) error {
+			gh := c.gh
+
+			prefix := "g"
+			if !c.time.IsZero() {
+				prefix += fmt.Sprintf(".Context(%d)", common.UnixMillis(c.time))
+			}
+
+			umd := api.NewUserMetadata("G.V().Has('Name', 'br-umd', 'Type', 'ovsbridge')", "testKey", "testValue")
+
+			c.client.Create("usermetadata", &umd)
+
+			_, err := gh.GetNode(prefix + ".V().Has('UserMetadata.testKey', 'testValue')")
+			if err != nil {
+				return err
+			}
+
+			c.client.Delete("usermetadata", umd.ID())
+
+			node, err := gh.GetNode(prefix + ".V().Has('UserMetadata.testKey', 'testValue')")
```

Comment:
You have twice the same test here and line 865, is that normal ?

Reply:
yes, the second one is to confirm the user metadata is deleted properly or not.

Reply:
ok got it :)